devops/scripts/benchmarks/utils/unitrace.py

```diff
@@ -0,0 +1,182 @@
+    os.makedirs(options.unitrace_res_dir, exist_ok=True)
+    if not options.save_name:
+        raise ValueError(
+            "Unitrace requires a save name to be specified via --save option."
+        )
+    bench_dir = f"{options.unitrace_res_dir}/{options.save_name}_{unitrace_timestamp}"
+    os.makedirs(bench_dir, exist_ok=True)
+
+    unitrace_output = f"{bench_dir}/{name}_{unitrace_timestamp}"
+    unitrace_command = (
+        [
+            str(unitrace_bin),
+            "--call-logging",
+            "--host-timing",
+            "--device-timing",
+            "--chrome-sycl-logging",
+            "--chrome-call-logging",
+            "--chrome-kernel-logging",
+            "--output",
+            unitrace_output,
+        ]
+        + extra_unitrace_opt
+        + command
+    )
+    if options.verbose:
+        print(f"Unitrace cmd: {' '.join(unitrace_command)}")
+
+    return bench_dir, unitrace_output, unitrace_command
+
+
+def handle_unitrace_output(bench_dir, unitrace_output, timestamp):
+    # Handle unitrace_output.{pid} logs: rename to unitrace_output (remove pid)
+    for f in os.listdir(bench_dir):
+        if f.startswith(os.path.basename(unitrace_output) + "."):
+            parts = f.rsplit(".", 1)
+            if (
+                len(parts) == 2
+                and parts[1].isdigit()
+                and os.path.isfile(os.path.join(bench_dir, f))
+            ):
+                src = os.path.join(bench_dir, f)
+                dst = os.path.join(bench_dir, os.path.basename(unitrace_output))
+                shutil.move(src, dst)
+                if options.verbose:
+                    print(f"Renamed {src} to {dst}")
+                break
+
+    # Handle {name}.{pid}.json files in cwd: move and rename to {self.name()}_{timestamp}.json
+    pid_json_files = []
+    for f in os.listdir(options.benchmark_cwd):
+        parts = f.split(".")
+        l = len(parts)
+        if len(parts) >= 3 and parts[l - 1] == "json" and parts[l - 2].isdigit():
+            pid_json_files.append(f)
+
+    if len(pid_json_files) == 1:
+        dst = f"{unitrace_output}.json"
+    else:
+        print(
+            f"Warning: Found {len(pid_json_files)} files matching the pattern. Expected 1."
+        )
+        # Find the newest file by modification time
+        newest_file = max(
+            pid_json_files,
+            key=lambda f: os.path.getmtime(os.path.join(options.benchmark_cwd, f)),
+        )
+        dst = f"{unitrace_output}.json"
+        for f in pid_json_files:
+            if f != newest_file:
+                os.remove(os.path.join(options.benchmark_cwd, f))
+                if options.verbose:
+                    print(f"Removed extra file {f}")
+
+    shutil.move(os.path.join(options.benchmark_cwd, pid_json_files[0]), dst)
+    if options.verbose:
+        print(f"Moved {pid_json_files[0]} to {dst}")
+
+    # Prune old unitrace directories
+    prune_unitrace_dirs(options.unitrace_res_dir, FILECNT=5)
```

Comment:
You use this only here. Why bother with defining a default value of an argument that you are going to override anyway?